dockstore-webservice/src/main/java/io/dockstore/webservice/core/Entry.java

```diff
@@ -306,6 +306,28 @@ public abstract class Entry<S extends Entry, T extends Version> implements Compa
     @Transient
     private Set<OrcidAuthor> orcidAuthors = new HashSet<>();
 
+    @JsonIgnore
+    @Column(nullable = true, columnDefinition = "varchar(32)")
+    @Schema(description = "The visibility of the Git repo")
+    @Enumerated(EnumType.STRING)
+    private GitVisibility gitVisibility;
+
+
+    public enum GitVisibility {
+        /**
+         * A private repo
+         */
+        PRIVATE,
+        /**
+         * A public repo
+         */
+        PUBLIC,
+        /**
+         * Unable to determine visibility
+         */
+        UNKNOWN
```

Comment:
May be answered later, is there a distinction between unknown unknown (i.e. before checking) and known unknown (i.e. after checking)?

Reply:
Looks like it would be the null value, could default to something more descriptive

Reply:
My thinking is that `null` is never checked, `UNKNOWN` means we checked but cannot determine if it's private or non-existent, most likely due to lack of permissions. I'll change the name and update the JavaDoc

Reply:
Having thought a bit more about this, one reason I dislike using `null` in these kinds of situations is that `null` can also mean uninitialized or yet to be loaded. So I kinda like an explicit enum value in these kinds of situations, kinda like why `Optional` exists

Reply:
> Having thought a bit more about this, one reason I dislike using `null` in these kinds of situations is that `null` can also mean uninitialized or yet to be loaded. So I kinda like an explicit enum value in these kinds of situations, kinda like why `Optional` exists

I started to to make it a non-nullable column, but on further reflection, decided not to. I hopefully made the enums clearer.

The reason I want to keep `null` is because of hosted workflows. For hosted, the field is not applicable, and `null` seems a better way to represent it.

I could create a `NOT_APPLICABLE` enum value, but then I'd still need a value for Bitbucket/GitLab as well, as well as GitHub based entries we haven't tried to determine the visiblity of.